src/Raven.Server/Documents/TimeSeries/TimeSeriesValuesSegment.cs

```diff
@@ -569,6 +580,13 @@ public Span<byte> AsSpan()
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool IsValid(ulong status, long value)
+        {
+            // valid = live and not NaN
+            return status == Live && (value & 0x7FFFFFFFFFFFFFFF) > 0x7FF0000000000000 == false;
```

Comment:
wouldn't it be better to use `double.IsNaN(Convert.LongToDoubleBits(value))` ?

Reply:
it is the impl of `double.IsNaN`, so I avoid the double conversion (from `long` to `double` and then back from `double` to `long`)